### PR TITLE
Added more error handling, prototype user object, and user lookup

### PR DESCRIPTION
**Pull request by glesage**

### CODE REVIEW

File: cloud/main.js [hunk 1]
```diff
@@ -17,22 +15,38 @@ Parse.Cloud.define("incomingSMS", function(request, response) {
...
+  
+});
+
 
+// -- Receiving User Registration -- //
+
+Parse.Cloud.define("newUser", function(request, response) {
+  console.log("---");
+
+  var user = user_handler.makeAUser("gefthefrench@gmail.com", "Jeff", "French", "3128602305", "https://dl.dropboxusercontent.com/u/30415492/Resume_Geoffroy.pdf");
```
Comment (MrMaksimize) [inline cloud/main.js]: Haha probably shouldn't commit this :)

File: cloud/main.js [hunk 2]
```diff
@@ -17,22 +15,38 @@ Parse.Cloud.define("incomingSMS", function(request, response) {
...
+
+  if (!user)  response.error();
+  else        response.success();
+
+  if (user) user.register();
```
Comment (MrMaksimize) [inline cloud/main.js]: Ti would probably be simpler to condense line 37 + 39

File: cloud/user_handler.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+// -- Experts -- //
+var phone_handler = require('cloud/phone.js');
+var email_handler = require('cloud/email.js');
+
+
+// -- Public User Constructor -- //
+
+exports.makeAUser = function(email, firstName, lastName, phone, resume) {
+
+  if (!email || email.length < 1)          return null;
```
Comment (MrMaksimize) [inline cloud/user_handler.js]: Wait so why are we bringing this in if user reg is done already?

File: public/index.html [hunk 4]
```diff
@@ -0,0 +1,9 @@
+
+<html>
+  <head>
```
Comment (MrMaksimize) [inline public/index.html]: lol i deleted this file :)

**Comment by MrMaksimize** [comment]
Also it's to dev :)

Wait did we agree not to take this one?

File: cloud/main.js (hunk 1 above)
Comment (glesage) [inline cloud/main.js]: Right... all the user registration stuff you can ommit... I've already removed from the branch I'm workign on

File: cloud/main.js (hunk 2 above)
Comment (glesage) [inline cloud/main.js]: Meh... its cleaner that way :P 
I feel it communicates the execution flow better

File: public/index.html (hunk 4 above)
Comment (glesage) [inline public/index.html]: Darn, I thought I did too.. too many things to put in Gitignore

File: cloud/user_handler.js (hunk 3 above)
Comment (glesage) [inline cloud/user_handler.js]: contructor for USER is private...

**Comment by glesage** [comment]
yea, unless you want to just merge 'Main' cus thats rly the only thing that changed for the better...
But I can just make another PR to master soon so yea wtv

**Pull request closed without merging**